pkg/collector/utils.go

```diff
@@ -20,30 +20,40 @@ import (
 	"github.com/sustainable-computing-io/kepler/pkg/cgroup"
 	collector_metric "github.com/sustainable-computing-io/kepler/pkg/collector/metric"
 	"k8s.io/klog/v2"
+
+	"github.com/sustainable-computing-io/kepler/pkg/kubernetes"
 )
 
+// this function is only called with the watcher delayed to sync and update the container info or if the watcher is not enabled
 func (c *Collector) createContainersMetricsIfNotExist(containerID string, cGroupID, pid uint64, withCGroupID bool) {
 	if _, ok := c.ContainersMetrics[containerID]; !ok {
-		var err error
-		var podName, containerName string
-
-		// the acceletator package does not get the cgroup ID, then we need to verify if we can call the cgroup with cGroupID or not
-		podName, _ = cgroup.GetPodName(cGroupID, pid, withCGroupID)
-		containerName, _ = cgroup.GetContainerName(cGroupID, pid, withCGroupID)
-
+		// We feel the info with generic values because the watcher will eventually update it.
+		podName := c.systemProcessName
```

Comment:
we don't need to alloc a new var named as `podName` in each loop right?

Reply:
We replace the content when there is no watcher....
I am keeping this for now, but we can improve it later...